src/main/logging/README.md

```diff
@@ -0,0 +1,244 @@
+## Deploy Logging Pattern - EFK Stack
+
+A common Kubernetes logging pattern is a combination of Fluent, Elasticsearch, and Kibana which is known as *EFK Stack*. 
+
+`Fluent` is an open source and multi-platform log processor which collects data/log from different sources, aggregates and forwards them to multiple destinations and is fully compatible with Docker and Kubernetes environments. 
+
+`Elasticsearch` is a distributed open search and analytics engine for all types of data. 
+
+`Kibana` is an open source frontend application that sits on top of Elasticsearch, providing search and data visualization capabilities for data indexed in Elasticsearch.
+
+There are different methods to deploy EFK stack and here we provide two deployment methods, first deploy EFK using helm chart and deploy local elasticsearch on Kubernetes and second using managed elasticsearch (EKS and AWS ES).
+
+## EFK using local elasticsearch
+
+This solution deploys all EFK stack inside the k8s cluster locally. By enabling fluentd, helm_install script installs fluentd in EKS. It means after running helm_install all product pods will run fluentd which collects all log files and sends them to the fluentd aggregator container. 
+
+To complete the EFK stack we need to install elasticsearch cluster and kibana and manage to forward the aggregated datalog to elasticsearch by fluentd which is already installed. 
+
+To start first enable `fluentd` and set the `hostname` for elasticsearch in volues.yaml as follows:
+
+```yaml
+fluentd:
+   enabled: true
+   elasticsearch:
+     hostname: elasticsearch-master
+```
+
+Now fluentd tries to parse and send the data to elasticsearch but as it is not installed yet then data get lost. At this point we should install elasticsearch and kibana to complete EFK stack deployment.
```

Comment:
Shouldn't we start with the installation of the elasticsearch first? It doesn't have any dependency on the helm charts so I would assume installing it first is a better deal.

Reply:
Agreed, changed the order